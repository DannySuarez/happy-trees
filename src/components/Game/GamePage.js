import React from 'react';
import styles from './game.css';

export default function Game() {
  return (
    <div className={styles.GameContainer}>

    </div>
  );
}