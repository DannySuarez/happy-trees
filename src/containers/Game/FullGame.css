@import url('https://fonts.googleapis.com/css?family=Gloria+Hallelujah&display=swap');
.FullGame {
    width: 95vw;
    height: 95vh;
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    align-content: center;
    

    h1 {
        color: pink;
        text-shadow: 3px 3px 4px DimGray;
        margin-bottom: 3vw;
    }

    .FullBorder {
        width: 93vw;
        height: 93vh;
        display: flex;
        flex-direction: column;
        align-items: center;
        border: 5px solid LightPink; 
        background: Aquamarine;
        border-radius: 10px;
        margin-top: 1vw;
        padding-top: 2vw;
        padding-bottom:2vw;
        opacity: 0.9;
        
    }

    .Word {
        justify-content: center;
        text-align: center;
        display: flex;
        flex-direction: row;
        height: 2vh;
        font-family: 'Gloria Hallelujah', cursive;
        font-size: 1em;
        text-decoration: underline;
        margin-bottom: 5vh;
        text-shadow: 2px 2px 0px  #fff,
        -2px -2px 0px  #fff,
        1px -1px 0px  #fff,
        -2px 1px #fff;
    }

    .GameBorder {
        display: flex;
        justify-content: center;
        align-items: center;
        text-align: center;
        width: 300px;
        height: 300px;
        border: solid .3em LightPink;
        border-radius: 10px;
        margin: 1vw;
    }

    .GameContainer {
        display: flex;
        justify-content: center;
        margin: auto;
        width: 300px;
        height: 300px;
        background-color: white;
        border: solid .3em #ff66c4;
        opacity: 1.0;
        border-radius:10px;

        -webkit-box-shadow: 
        inset 0 0 8px  rgba(0,0,0,0.1),
              0 0 16px rgba(0,0,0,0.1),
              0px 0px 3px 3px LightGrey;  
      -moz-box-shadow: 
        inset 0 0 8px  rgba(0,0,0,0.1),
              0 0 16px rgba(0,0,0,0.1), 
              0px 0px 3px 3px LightGrey; 
      box-shadow: 
        inset 0 0 8px  rgba(0,0,0,0.1),
              0 0 16px rgba(0,0,0,0.1), 
              0px 0px 3px 3px LightGrey; 
      }

    .GameInput {
        margin: auto;
        justify-content: center;
        text-align: center;
        display: flex;
        flex-direction: column;
        width: 50vw;
    }
}

@media screen and (max-width: 320px) and (max-height: 568px){

    .FullGame {
        width: 95vw;
        height: 95vh;
        display: flex;
        flex-wrap: nowrap;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        align-content: center;
        
    
        h1 {
            color: pink;
            text-shadow: 3px 3px 4px DimGray;
            margin-bottom: 3vw;
        }
    
        .FullBorder {
            width: 93vw;
            height: 93vh;
            display: flex;
            flex-direction: column;
            align-items: center;
            border: 5px solid LightPink; 
            background: Aquamarine;
            border-radius: 10px;
            margin-top: 1vw;
            padding-top: 2vw;
            padding-bottom:2vw;
            opacity: 0.9;
            
        }
    
        .Word {
            justify-content: center;
            text-align: center;
            display: flex;
            flex-direction: row;
            height: 2vh;
            font-family: 'Gloria Hallelujah', cursive;
            font-size: 1em;
            text-decoration: underline;
            padding: 5px;
            text-shadow: 2px 2px 0px  #fff,
            -2px -2px 0px  #fff,
            1px -1px 0px  #fff,
            -2px 1px #fff;
        }
    
        .GameBorder {
            display: flex;
            justify-content: center;
            align-items: center;
            text-align: center;
            width: 275px;
            height: 275px;
            border: solid .3em LightPink;
            border-radius: 10px;
            margin: 1vw;
        }
    
        .GameContainer {
            display: flex;
            justify-content: center;
            margin: auto;
            width: 275px;
            height: 275px;
            background-color: white;
            opacity: 1.0;
            border-radius: 10px;
          }
    
        .GameInput {
            margin: auto;
            justify-content: center;
            text-align: center;
            display: flex;
            flex-direction: column;
            width: 50vw;
        }
    }
}
