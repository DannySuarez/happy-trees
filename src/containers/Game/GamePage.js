import React from 'react';
import PropTypes from 'prop-types';
import styles from './game.css';

import P5Wrapper from 'react-p5-wrapper';
import io from 'socket.io-client';
import { connect } from 'react-redux';
import sketch from '../../sketch/sketch';
import { beginListening, endListening, joinedGame, gameStarted, startNewRound } from '../../actions/socketActions';
import { getStrokes } from '../../selectors/drawingSelectors';
import { getUserNickname } from '../../selectors/authSelectors';
import { receiveStroke } from '../../actions/drawingActions';

import StatusBar from '../../components/gameInput/StatusBar';
import { getUserId } from '../../selectors/authSelectors';
import { getIsDrawing, getGameId, getRoundId, getIsPlaying } from '../../selectors/socketSelectors';
import GameInput from '../../components/gameInput/GameInput';

class GamePage extends React.Component {

  static propTypes ={
    startListening: PropTypes.func.isRequired,
    stopListening: PropTypes.func.isRequired,
    receiveStroke: PropTypes.func.isRequired,
    joinedGame: PropTypes.func.isRequired,
    gameStarted: PropTypes.func.isRequired,
    strokes: PropTypes.array.isRequired,
    userId: PropTypes.string.isRequired,
    isDrawing: PropTypes.bool.isRequired,
    gameId: PropTypes.string,
    roundId: PropTypes.string,
    nickname: PropTypes.string,
    isPlaying: PropTypes.bool.isRequired,
    startNewRound: PropTypes.func.isRequired
  }

  state = {
    cavasWidth: null,
    canvasHeight: null,
    time: null,
    guess: '',
  }

  socket = io('http://localhost:3000');

  componentDidMount() {
    const gameContainer = document.querySelector('#game-container');
    if(gameContainer && gameContainer.offsetWidth && gameContainer.offsetWidth) {
      this.setState({ 
        canvasWidth:  gameContainer.offsetWidth, 
        canvasHeight: gameContainer.offsetHeight 
      });
    }

    this.listenForGameEvents();
    this.socket.emit('find game');
  }

  listenForGameEvents = () => {
    this.props.startListening();

    this.socket.on('stroke', data => {
      this.props.receiveStroke(data);
    });

    this.socket.on('timer', ({ countdown, round }) => {
      console.log('round timer', countdown, round);
      this.setState({ time: countdown });
    });

    this.socket.on('joined game', gameId => this.props.joinedGame(gameId));

    this.socket.on('start game', (startRound) => {
      const { userId } = this.props;
      this.props.gameStarted(startRound, userId);
    });

    this.socket.on('correct answer', () => {
      console.log('someone made a correct answer');
    });

    this.socket.on('wrong answer', () => [
      console.log('someone made a wrong answer')
    ]);

    this.socket.on('intermission', ({ countdown }) => {
      console.log('intermission', countdown);
    });

    this.socket.on('new round', ({ round }) => {
      console.log('new round', round);
      this.props.startNewRound(round, this.props.userId);
    });

    this.socket.on('round over', () => {
      console.log('round over');
    });

    this.socket.on('game over', () => {
      console.log('game over');
    });
  }

  componentWillUnmount() {
    this.socket.removeListener('stroke');
    this.socket.emit('disconnect');
    this.props.stopListening();
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  emitStroke = (data) => {
    const { gameId } = this.props;
    this.props.receiveStroke(data);
    this.socket.emit('stroke', { data, gameId });
  }

  emitAnswer = (e) => {
    e.preventDefault();
    const { gameId, roundId } = this.props;
    const { guess } = this.state;
    this.socket.emit('answer', {
      answer: guess,
      roundId,
      gameId,
      currentRoundNumber: 1
    });
    this.setState({ guess: '' });
  }
  
  render() {
    const { canvasWidth, canvasHeight, time, guess } = this.state;
    const { isDrawing, nickname, strokes, isPlaying } = this.props;
    
    return (
      <>
        <StatusBar nickname={nickname} time={time} />
      
        <div id="game-container" className={styles.GameContainer}>
          <P5Wrapper 
            sketch={sketch} 
            color={'#000000'} 
            canvasWidth={canvasWidth} 
            canvasHeight={canvasHeight}
            emitStroke={this.emitStroke}
            strokes={strokes}
            isDrawing={isDrawing}
          />
        </div>
        {isPlaying && !isDrawing && <GameInput 
          guesses={3}
          guess={guess}
          handleSubmit={this.emitAnswer}
          handleChange={this.handleChange}
        />}
        
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  strokes: getStrokes(state),
  userId: getUserId(state),
  isDrawing: getIsDrawing(state),
  gameId: getGameId(state),
  roundId: getRoundId(state),
  nickname: getUserNickname(state),
  isPlaying: getIsPlaying(state)
});

const mapDispatchToProps = dispatch => ({
  startListening: () => dispatch(beginListening()),
  stopListening: () => dispatch(endListening()),
  receiveStroke: (data) => dispatch(receiveStroke(data)),
  joinedGame: (gameId) => dispatch(joinedGame(gameId)),
  gameStarted: (startRound, userId) => dispatch(gameStarted(startRound, userId)),
  startNewRound: (round, userId) => dispatch(startNewRound(round, userId))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GamePage);
