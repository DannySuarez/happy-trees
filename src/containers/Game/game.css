.GameContainer {
  margin: auto;
  margin-top: 5%;
  width: 90vw;
  height: 70vh;
  background-color: #ffffff;
  border: solid #000000 .5em;
  border-radius: 12px;
}