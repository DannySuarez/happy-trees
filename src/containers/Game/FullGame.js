import React from 'react';
import StatusBar from '../../components/gameInput/StatusBar';
import GamePage from './GamePage';
import GameInput from '../../components/gameInput/GameInput';
import styles from './FullGame.css';


const FullGame = () => {

  return (
    <div className={styles.FullGame}>
      <h1>Happy Trees</h1>
      <StatusBar />
      <div className={styles.Word}>
        <h3>W o r d</h3>
      </div>
      <GamePage />
      <div className={styles.GameInput}>
        <GameInput />
      </div>
    </div>
  );
};

export default FullGame;
