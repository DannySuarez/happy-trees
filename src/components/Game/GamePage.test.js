import React from 'react';
import { shallow } from 'enzyme';
import Game from './Gamepage';

describe('Game component', () => {
  it('renders Game page', () => {
    const wrapper = shallow(<Game />);
    expect(wrapper).toMatchSnapshot();
  });
});
