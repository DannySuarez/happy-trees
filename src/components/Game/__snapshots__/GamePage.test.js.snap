// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders Game page 1`] = `
<div
  className="GameContainer"
/>
`;
